pytorch_lightning/overrides/base.py

```diff
@@ -19,25 +21,61 @@
 from pytorch_lightning.utilities.device_dtype_mixin import DeviceDtypeModuleMixin
 
 
-class _LightningModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
+class _LightningPrecisionModuleWrapperBase(torch.nn.Module):
```

Comment:
Do you also want to do a fully fp16 based version (instead of only the AMP thing pytorch and apex provide) or why did you also introduce a base class for this? If not, we probably don't need the base class as of now.

Reply:
The reason to have a base wrapper is that the `unwrap_lightning_module` method needs to know how to unwrap both wrapper types. The other option would be to just make the double precision wrapper extend `_LightningModuleWrapperBase` and make the unwrap method recursive